js/pro/gemini.js

```diff
@@ -0,0 +1,664 @@
+        //
+        //     {
+        //         type: 'l2_updates',
+        //         symbol: 'BTCUSD',
+        //         changes: [
+        //             [ 'buy', '22252.37', '0.02' ],
+        //             [ 'buy', '22251.61', '0.04' ],
+        //             [ 'buy', '22251.60', '0.04' ],
+        //             // some asks as well
+        //         ],
+        //         trades: [
+        //             { type: 'trade', symbol: 'BTCUSD', event_id: 122258166738, timestamp: 1655330221424, price: '22269.14', quantity: '0.00004473', side: 'buy' },
+        //             { type: 'trade', symbol: 'BTCUSD', event_id: 122258141090, timestamp: 1655330213216, price: '22250.00', quantity: '0.00704098', side: 'buy' },
+        //             { type: 'trade', symbol: 'BTCUSD', event_id: 122258118291, timestamp: 1655330206753, price: '22250.00', quantity: '0.03', side: 'buy' },
+        //         ],
+        //         auction_events: [
+        //             {
+        //                 type: 'auction_result',
+        //                 symbol: 'BTCUSD',
+        //                 time_ms: 1655323200000,
+        //                 result: 'failure',
+        //                 highest_bid_price: '21590.88',
+        //                 lowest_ask_price: '21602.30',
+        //                 collar_price: '21634.73'
+        //             },
+        //             {
+        //                 type: 'auction_indicative',
+        //                 symbol: 'BTCUSD',
+        //                 time_ms: 1655323185000,
+        //                 result: 'failure',
+        //                 highest_bid_price: '21661.90',
+        //                 lowest_ask_price: '21663.79',
+        //                 collar_price: '21662.845'
+        //             },
+        //         ]
+        //     }
+        //
+        this.handleOrderBook (client, message);
+        this.handleTrades (client, message);
+    }
+
+    async watchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name gemini#fetchOrders
+         * @description watches information on multiple orders made by the user
+         * @see https://docs.gemini.com/websocket-api/#order-events
+         * @param {string|undefined} symbol unified market symbol of the market orders were made in
+         * @param {int|undefined} since the earliest time in ms to fetch orders for
+         * @param {int|undefined} limit the maximum number of  orde structures to retrieve
+         * @param {object} params extra parameters specific to the gemini api endpoint
+         * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        const url = this.urls['api']['ws'] + '/v1/order/events?eventTypeFilter=initial&eventTypeFilter=accepted&eventTypeFilter=rejected&eventTypeFilter=fill&eventTypeFilter=cancelled&eventTypeFilter=booked';
```

Comment:
@pcriadoperez `watchOrders` is throwing this network error: 
![image](https://user-images.githubusercontent.com/43336371/204022843-0902405c-441f-444f-89e3-dbfa01fdd9c1.png)


can you please take a look?

Reply:
I was getting the same issue from using `fetchMarkets` I opened this PR to solve the problem https://github.com/ccxt/ccxt/pull/15862